app/src/main/java/de/tum/in/tumcampusapp/component/tumui/grades/GradesFragment.kt

```diff
@@ -159,13 +166,47 @@ class GradesFragment : FragmentForAccessingTumOnline<ExamList>(
             data = BarData(set)
             setFitBars(true)
 
+            // only label grades that are associated with at least one grade
+            data.setValueFormatter { value, _, _, _ ->
+                if (value > 0) {
+                    "${value.toInt()}"
+                } else {
+                    ""
+                }
+            }
+
             xAxis.apply {
                 granularity = 1f
                 setValueFormatter { value, _ -> grades[value.toInt()] }
             }
 
             description = null
             setTouchEnabled(false)
+
+            // set up elements for bar-charts legend. Libraries default legend would contain too many grades, we only want common ones
+            // this might be refined
+            val newLegend = ArrayList<LegendEntry>()
+            for (i: Int in 0..4) {
+                for (j: Int in listOf(0, 3, 7)) {
+                    newLegend.add(
+                            LegendEntry(
+                                    null, Legend.LegendForm.DEFAULT, 10f, 10f, null,
+                                    ColorTemplate.rgb(resources.getString(GRADE_COLORS[10 * i + j]))
+                            )
+                    )
+                    if (i == 4) {
```

Comment:
that iteration logic isn't too obvious. I think just hardcoding it in an `DEFAULT_GRADE_COLORS` array might be simpler

Reply:
Without the legend we won't need to worry about all this mess :+1: